### ISSUE TITLE
Upserting a long document with `front` timeouting does not complete on `core` side

### ISSUE THREAD

**Issue by spolu** [dust-tt/dust#413]
Logs stop while embedding, which is unexpected, need to investigate locally

**Comment by spolu** [comment]
cc @fontanierh

**Comment by spolu** [comment]
Was able to reproduce the upsertion with the following code:
```
import fs from "fs";

const CORE_API = "http://127.0.0.1:3001";

const { OPENAI_API_KEY } = process.env;

const main = async () => {
  const pRes = await fetch(`${CORE_API}/projects`, {
    method: "POST",
  });

  const p = await pRes.json();
  // console.log(p);
  let projectId = p.response.project.project_id;
  console.log(`Created project: ${projectId}`);

  const dsRes = await fetch(`${CORE_API}/projects/${projectId}/data_sources`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      data_source_id: "ds-test",
      config: {
        provider_id: "openai",
        model_id: "text-embedding-ada-002",
        splitter_id: "base_v0",
        max_chunk_size: 256,
        use_cache: false,
      },
      credentials: {
        OPENAI_API_KEY,
      },
    }),
  });

  const ds = await dsRes.json();
  // console.log(ds);
  let dataSourceId = ds.response.data_source.data_source_id;
  console.log(`Created data source: ${dataSourceId}`);

  // read local file content.txt
  const content = fs.readFileSync("./content.txt", "utf8");
  console.log(`Read content.txt: ${content.length} chars`);

  const dRes = await fetch(
    `${CORE_API}/projects/${projectId}/data_sources/${dataSourceId}/documents`,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        document_id: "test-large",
        timestamp: Date.now(),
        tags: [],
        source_url: null,
        text: content,
        credentials: {
          OPENAI_API_KEY,
        },
      }),
    }
  );
  const d = await dRes.json();
  console.log(d);
};

main()
  .then(() => {
    console.error("\x1b[32m%s\x1b[0m", `Done`);
    process.exit(0);
  })
  .catch((err) => {
    console.error("\x1b[31m%s\x1b[0m", `Error: ${err.message}`);
    console.log(err);
    process.exit(1);
  });
```

**Comment by spolu** [comment]
Everything worked locally :thinking:

**Comment by spolu** [comment]
I did get one error at some point that led to a 500 because the TLS handshake with OpenAI failed, but there was a log in the output of `core`.

I then went back to production machines to look for the full logs and find a similar response logged but could not find one.

Given that it worked quite well locally, this does not seem too urgent. Will close the issue and as mitigation will work on making `core` push logs to datadog.
